pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/StarTreeBuilderUtils.java

```diff
@@ -83,6 +85,27 @@ public static List<StarTreeV2BuilderConfig> generateBuilderConfigs(@Nullable Lis
     return builderConfigs;
   }
 
+  public static List<StarTreeV2BuilderConfig> generateBuilderConfigs(@Nullable List<StarTreeIndexConfig> indexConfigs,
+      boolean enableDefaultStarTree, Schema schema, JsonNode segmentMetadata) {
+    List<StarTreeV2BuilderConfig> builderConfigs = new ArrayList<>();
+    if (indexConfigs != null) {
+      for (StarTreeIndexConfig indexConfig : indexConfigs) {
+        StarTreeV2BuilderConfig builderConfig = StarTreeV2BuilderConfig.fromIndexConfig(indexConfig);
+        if (!builderConfigs.contains(builderConfig)) {
+          builderConfigs.add(builderConfig);
+        }
+      }
+    }
+    if (enableDefaultStarTree) {
```

Comment:
What's the current behavior if

1. custom startree config is added
2. enableDefaultStarTree is set to true

Are we adding default startree or default startree + custom startreee?

Reply:
yes that is the current behaviour right now.

Existing `generateBuilderConfigs(@Nullable List<StarTreeIndexConfig> indexConfigs,
      boolean enableDefaultStarTree, SegmentMetadata segmentMetadata)` hints at that as well.